tests/test_cli.py

```diff
@@ -19,10 +22,13 @@ def test_cli_help():
     assert "Export a dataset to MBTiles (version 1.1)" in result.output
 
 
-def test_nodata_validation():
-    """Insufficient nodata definition leads to BadParameter"""
-    with pytest.raises(click.BadParameter):
-        validate_nodata(0, None, None)
+@mock.patch("mbtiles.scripts.cli.rasterio")
+def test_dst_nodata_validation(rio):
+    """--dst-nodata requires source nodata in some form"""
+    rio.open.return_value.__enter__.return_value.profile.get.return_value = None
+    runner = CliRunner()
+    result = runner.invoke(main_group, ["mbtiles", "--dst-nodata", "0", "in.tif", "out.mbtiles"])
+    assert result.exit_code == 2
```

Comment:
This test required a new test dataset or a fake dataset. I went with the latter option. The two tricky things about mocking `rasterio.open`: 1) the module importing it needs to be imported before tests are run for `patch()` to work, and 2) we need to keep in mind that we may be entering a managed context, hence `__enter__` in the chain.